### PR TITLE
Fix potential endless loop on CI systems for iTC team selection

### PR DESCRIPTION
**Pull request by KrauseFx**
Related https://github.com/fastlane/fastlane/pull/8827
Fixes https://github.com/fastlane/fastlane/issues/8824

### CODE REVIEW

File: spaceship/lib/spaceship/tunes/tunes_client.rb [hunk 1]
```diff
@@ -86,6 +86,12 @@ def select_team
           puts "#{i + 1}) \"#{team['contentProvider']['name']}\" (#{team['contentProvider']['contentProviderId']})"
         end
 
+        unless Spaceship::Client::UserInterface.interactive?
+          puts "Multiple teams found on iTunes Connect, Your Terminal is running in non-interactive mode! Cannot continue from here."
+          puts "Please check that you set FASTLANE_ITC_TEAM_ID or FASTLANE_ITC_TEAM_NAME to the right value."
+          raise "Multiple iTunes Connect Teams found; unable to choose, terminal not ineractive!"
```
Comment (ohayon) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: Can this be a `user_error`?

File: spaceship/spec/tunes/tunes_client_spec.rb [hunk 2]
```diff
@@ -55,4 +55,13 @@
       end
     end
   end
+
+  describe "CI" do
+    it "crashes when running in non-interactive shell" do
+      expect(FastlaneCore::Helper).to receive(:ci?).and_return(true)
+      provider = { 'contentProvider' => { 'name' => 'Tom', 'contentProviderId' => 1234 } }
+      allow(subject).to receive(:teams).and_return([provider, provider]) # pass it twice, to call the team selection
+      expect { subject.select_team }.to raise_error "Multiple iTunes Connect Teams found; unable to choose, terminal not ineractive!"
```
Comment (ohayon) [inline spaceship/spec/tunes/tunes_client_spec.rb]: raise a user error?

**Review by ohayon — CHANGES_REQUESTED** [review]
This is 💯 , just wanted to talk about whether or not this should raise a user error?

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | File|
--- | ---|
Match Match::Encrypt raises an exception if invalid password is passed | <a href='https://github.com/fastlane/fastlane/blob/3027d3bbd18517badff8becd3e7edee9d159e0b3/./match/spec/encrypt_spec.rb'>encrypt_spec.rb</a>|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: spaceship/lib/spaceship/tunes/tunes_client.rb (hunk 1 above)
Comment (KrauseFx) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: I don't think so, as _spaceship_ doesn't have access to FastlaneCore, including UserError

Comment (ohayon) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: Ah, so this is going to bubble up as a regular crash right?

**Review by ohayon — APPROVED** [review]

File: spaceship/lib/spaceship/tunes/tunes_client.rb (hunk 1 above)
Comment (KrauseFx) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: Yes :)

**Pull request merged**